src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -781,8 +782,8 @@ private String getCheckoutEffectiveCredentials() {
                         URLEncoder.encode(Constants.R_HEADS + head.getName(), "UTF-8");
                 title = null;
             }
-            result.add(new BitbucketLink("icon-bitbucket-branch", serverUrl + "/" + branchUrl));
-            result.add(new ObjectMetadataAction(title, null, serverUrl+"/"+branchUrl));
+            result.add(new BitbucketLink("icon-bitbucket-branch", serverUrl + "/" + URIUtil.encodePath(branchUrl)));
```

Comment:
@alexbrjo One thing to double check here: confirm that serverUrl can't be already URL-encoded. 

May be worth escaping at the repoOwner/repository level rather than once you have a branchUrl.